### DISCUSSION TITLE
[Bug]: Reports exporting not working

### DISCUSSION THREAD

**Discussion by danielponte** [mbentley/docker-omada-controller#448] [Help] [answered]
### Controller Version

v5.14.26

### Describe the Bug

Trying to export a report to PDF results in this error:

Failed to export report and possible reasons are:
1. Insufficient space. Please make sure your PC has the free memory space of over 1.2 GB.
2. Damaged software. It is recommended to reinstall the controller.
3. Privilege restriction of the file path. It is recommended to lift the privilege restriction first.

Exporting to .CSV is fine

### Expected Behavior

Download the PDF :)

### Steps to Reproduce

Go into the reports section and click on the top right icon to export

### How You're Launching the Container

```plain
Running exactly as the example on the site:


docker run -d \
  --name omada-controller \
  --stop-timeout 60 \
  --restart unless-stopped \
  --ulimit nofile=4096:8192 \
  -p 8088:8088 \
  -p 8043:8043 \
  -p 8843:8843 \
  -p 27001:27001/udp \
  -p 29810:29810/udp \
  -p 29811-29816:29811-29816 \
  -e MANAGE_HTTP_PORT=8088 \
  -e MANAGE_HTTPS_PORT=8043 \
  -e PGID="508" \
  -e PORTAL_HTTP_PORT=8088 \
  -e PORTAL_HTTPS_PORT=8843 \
  -e PORT_ADOPT_V1=29812 \
  -e PORT_APP_DISCOVERY=27001 \
  -e PORT_DISCOVERY=29810 \
  -e PORT_MANAGER_V1=29811 \
  -e PORT_MANAGER_V2=29814 \
  -e PORT_TRANSFER_V2=29815 \
  -e PORT_RTTY=29816 \
  -e PORT_UPGRADE_V1=29813 \
  -e PUID="508" \
  -e SHOW_SERVER_LOGS=true \
  -e SHOW_MONGODB_LOGS=false \
  -e SSL_CERT_NAME="tls.crt" \
  -e SSL_KEY_NAME="tls.key" \
  -e TZ=Etc/UTC \
  -v omada-data:/opt/tplink/EAPController/data \
  -v omada-logs:/opt/tplink/EAPController/logs \
  mbentley/omada-controller
```

### Container Logs

```plain
Nothing happens on the log when the UI spits out the error. Nada.
```

### MongoDB Logs

```plain
Same as omada log.
```

**Comment by mbentley** [comment, marked as answer]
You need to use the image with chromium as that is used the generate the PDFs.  See [the tags section of the README](https://github.com/mbentley/docker-omada-controller?tab=readme-ov-file#tags-with-chromium) as there is a specific tag.  It adds quite a bit of size to the image which is why there is a separate one.

**Reply by danielponte** [reply]
> You need to use the image with chromium as that is used the generate the PDFs. See [the tags section of the README](https://github.com/mbentley/docker-omada-controller?tab=readme-ov-file#tags-with-chromium) as there is a specific tag. It adds quite a bit of size to the image which is why there is a separate one.

How can I pull an arm64 chromium version? Im trying to use this on rpi4.

Thanks.

**Reply by mbentley** [reply]
Unfortunately there isn't one.  I'm having to use a PPA and that PPA only has the amd64 version.  The reason for the PPA is because Ubuntu 20.04 has Chromium [but it's a snap](https://packages.ubuntu.com/search?keywords=chromium&searchon=names&suite=focal&section=all) instead of just normal binaries so it doesn't work inside the container.  [This is the thread](https://github.com/mbentley/docker-omada-controller/issues/214) where I originally implemented adding Chromium.

**Reply by danielponte** [reply]
> Unfortunately there isn't one. I'm having to use a PPA and that PPA only has the amd64 version. The reason for the PPA is because Ubuntu 20.04 has Chromium [but it's a snap](https://packages.ubuntu.com/search?keywords=chromium&searchon=names&suite=focal&section=all) instead of just normal binaries so it doesn't work inside the container. [This is the thread](https://github.com/mbentley/docker-omada-controller/issues/214) where I originally implemented adding Chromium.

Aww.. I've been trying to do this on a Proxmox (LCX containers).. hence it doesnt work. Need to  find a PPA with an arm64 version. Mind sharing what PPA you're using for the amd64 one?

**Reply by mbentley** [reply]
Well there is [this](https://askubuntu.com/a/1206153) which I just found when searching again - basically saying to install it from the Debian buster repos.  That's honestly kind of a nightmare though but it might work.  I may play with it to see if it even works.

The PPA is [listed in the Dockerfile](https://github.com/mbentley/docker-omada-controller/blob/master/Dockerfile.chromium#L8).

**Reply by mbentley** [reply]
I am trying to get this working with the debian packages.  I just tried to generate a report after getting it installed but it failed.  I'll have to do more debugging once I have a chance.

**Reply by mbentley** [reply]
Ah, so you're probably out of luck.  They use an application in `/opt/tplink/EAPController/bin/topdf` to generate the PDFs.  This is an x86-64 only binary as you can see from the output of `file`:

```
# file /opt/tplink/EAPController/bin/topdf
/opt/tplink/EAPController/bin/topdf: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, BuildID[sha1]=3cb1a5d531f2b89baef69d5bbb4f0688959a572e, with debug_info, not stripped
```

I would suggest creating a forum post on the [TP-Link forums](https://community.tp-link.com/en/business/forum/topic/569686?replyId=1111892) for feature request to support non-amd64 (x86_64) PDF generation.

**Comment by danielponte** [comment]
Thank you and sorry about posting a bug... will try that!
